cmd/zoekt-dynamic-indexserver/main.go

```diff
@@ -132,7 +137,23 @@ func (s *indexServer) serveHealthCheck(w http.ResponseWriter, r *http.Request) {
 	// Nothing to do. Just return 200
 }
 
+var (
+	promRegistry        = prometheus.NewRegistry()
+	metricRequestsTotal = promauto.With(promRegistry).NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "zoekt_dynamic_indexserver_requests_total",
+			Help: "Total number of HTTP requests by status code and method.",
+		},
+		[]string{"code", "method"},
+	)
+)
+
+func (s *indexServer) serveMetrics(w http.ResponseWriter, r *http.Request) {
+	promhttp.HandlerFor(promRegistry, promhttp.HandlerOpts{Registry: promRegistry}).ServeHTTP(w, r)
+}
+
 func (s *indexServer) serveIndex(w http.ResponseWriter, r *http.Request) {
+	method := "index"
```

Comment:
I think `method` has a well defined meaning in the context of HTTP requests. We should get it like:

```suggestion
	method := r.Method
```

Then if we add `path` like in https://github.com/sourcegraph/zoekt/pull/541/files#r1112478338 it should cover what you need.

It is kinda weird though with `method` because we actually just ignore this in our request routing anyway. But eventually it might start to have a meaning (eg. `GET /index` might give us details about what's in the index?)

Reply:
That totally makes sense. I've changed the code so that we have `method` and `path` 👍